chandra_aca/star_probs.py

```diff
@@ -227,24 +247,167 @@ def acq_success_prob(date=None, t_ccd=-10.0, mag=10.0, color=0.6, spoiler=False,
+    :param halfwidth: search box halfwidth (arcsec, default=120, float or ndarray)
+    :param probit: if True then return Probit(p_success). Default=False
+    :param model: Model name, e.g. 'grid-floor-2018-11'
+
+    :returns: Acquisition success probability(s)
+
+    """
+    # Info about model is cached in GRID_FUNCS
+    if model not in GRID_FUNCS:
+        from astropy.io import fits
+        from scipy.interpolate import RegularGridInterpolator
+
+        # Read the model file and put into local vars
+        filename = os.path.join(STAR_PROBS_DATA_DIR, model) + '.fits.gz'
+        if not os.path.exists(filename):
+            raise IOError('model file {} does not exist'.format(filename))
+
+        hdus = fits.open(filename)
+        hdu0 = hdus[0]
+        probit_p_fail_no_1p5 = hdus[1].data
+        probit_p_fail_1p5 = hdus[2].data
+
+        hdr = hdu0.header
+        grid_mags = np.linspace(hdr['mag_lo'], hdr['mag_hi'], hdr['mag_n'])
+        grid_t_ccds = np.linspace(hdr['t_ccd_lo'], hdr['t_ccd_hi'], hdr['t_ccd_n'])
+        grid_halfws = np.linspace(hdr['halfw_lo'], hdr['halfw_hi'], hdr['halfw_n'])
+
+        # Sanity checks on model data
+        assert probit_p_fail_no_1p5.shape == (len(grid_mags),
+                                              len(grid_t_ccds),
+                                              len(grid_halfws))
+        assert probit_p_fail_1p5.shape == probit_p_fail_no_1p5.shape
+
+        # Generate the 3-d linear interpolation functions
+        func_no_1p5 = RegularGridInterpolator(points=(grid_mags, grid_t_ccds, grid_halfws),
+                                              values=probit_p_fail_no_1p5)
+        func_1p5 = RegularGridInterpolator(points=(grid_mags, grid_t_ccds, grid_halfws),
+                                           values=probit_p_fail_1p5)
+        mag_lo = hdr['mag_lo']
+        mag_hi = hdr['mag_hi']
+        t_ccd_lo = hdr['t_ccd_lo']
+        t_ccd_hi = hdr['t_ccd_hi']
+        halfw_lo = hdr['halfw_lo']
+        halfw_hi = hdr['halfw_hi']
+
+        GRID_FUNCS[model] = {'filename': filename,
+                             'func_no_1p5': func_no_1p5,
+                             'func_1p5': func_1p5,
+                             'mag_lo': mag_lo,
+                             'mag_hi': mag_hi,
+                             't_ccd_lo': t_ccd_lo,
+                             't_ccd_hi': t_ccd_hi,
+                             'halfw_lo': halfw_lo,
+                             'halfw_hi': halfw_hi}
+    else:
+        gfm = GRID_FUNCS[model]
+        func_no_1p5 = gfm['func_no_1p5']
+        func_1p5 = gfm['func_1p5']
+        mag_lo = gfm['mag_lo']
+        mag_hi = gfm['mag_hi']
+        t_ccd_lo = gfm['t_ccd_lo']
+        t_ccd_hi = gfm['t_ccd_hi']
+        halfw_lo = gfm['halfw_lo']
+        halfw_hi = gfm['halfw_hi']
+
+    # Make sure inputs are within range of gridded model
+    mag = clip_and_warn('mag', mag, mag_lo, mag_hi, model)
```

Comment:
I'm a little uneasy about altering the input data this way, but it makes sense and the warning seems fine.

Reply:
In this case (where clipping actually does something), the input data are invalid for the model.  So the choice is either to throw an exception, clip and warn, or extrapolate.  Of those I think that clip and warn is the best.